quickwit/quickwit-config/src/storage_config.rs

```diff
@@ -140,6 +143,15 @@ impl StorageConfigs {
                 _ => None,
             })
     }
+
+    pub fn find_cache(&self) -> Option<&CacheStorageConfig> {
+        self.0
+            .iter()
+            .find_map(|storage_config| match storage_config {
+                StorageConfig::Cache(cache_storage_config) => Some(cache_storage_config),
+                _ => None,
+            })
```

Comment:
personal nitpick. I tend to prefer to avoid rare combinators.

```rust
for storage_config in &self.0 {
    if let StorageConfig::Cache(cache_storage_config) = storage_config {
        return Some(cache_storage_config);
    }
}
None
```

Reply:
This code is repeated 5 times in this file. We should probably replace all occurrences then.